tests/lib/e2e/buckets/files.e2e-spec.ts

```diff
@@ -0,0 +1,400 @@
+          .post('/buckets')
+          .set('Authorization', getAuth(testUser))
+
+        // Arrange: start the upload
+        const response = await testServer.post(`/v2/buckets/${bucketId}/files/start`)
+          .set('Authorization', getAuth(testUser))
+          .send({ uploads: [{ index: 0, size: 1000, }] })
+
+        const { uploads: [upload] } = response.body;
+
+        // Act: finish the upload
+        const index = crypto.randomBytes(32).toString('hex');
+        const responseComplete = await testServer.post(`/v2/buckets/${bucketId}/files/finish`)
+          .set('Authorization', getAuth(testUser))
+          .send({
+            index,
+            shards: [{ hash: crypto.randomBytes(20).toString('hex'), uuid: upload.uuid, }],
+          });
+
+        // Assert
+        expect(responseComplete.status).toBe(200);
+
+        const body = responseComplete.body;
+
+        expect(body.bucket).toEqual(bucketId);
+        expect(body.created).toBeDefined();
+        expect(body.filename).toBeDefined();
+        expect(body.id).toBeDefined();
+        expect(body.index).toEqual(index);
+        expect(body.mimetype).toBeDefined();
+        expect(body.renewal).toBeDefined();
+        expect(body.size).toBeGreaterThan(0);
+        expect(typeof body.size).toBe('number');
+        expect(body.version).toBe(2);
+      });
+
+    })
+
+    describe('Downloading a file', () => {
+
+      it('When a user wants to download a file, it should get a list of links for each file part', async () => {
+        // Arrange: Create a bucket
+        const { body: { id: bucketId } } = await testServer
+          .post('/buckets')
+          .set('Authorization', getAuth(testUser))
+
+        // Arrange: start the upload
+        const { body: { uploads } } = await testServer.post(`/v2/buckets/${bucketId}/files/start`)
+          .set('Authorization', getAuth(testUser))
+          .send({ uploads: [{ index: 0, size: 1000, }, { index: 1, size: 10000, },], })
+
+        // Arrange: finish the upload
+        const index = crypto.randomBytes(32).toString('hex');
+        const { body: file } = await testServer.post(`/v2/buckets/${bucketId}/files/finish`)
+          .set('Authorization', getAuth(testUser))
+          .send({
+            index,
+            shards: (uploads as any[]).map(upload => ({ hash: crypto.randomBytes(20).toString('hex'), uuid: upload.uuid, })),
+          });
+
+
+        // Act: download the file
+        const response = await testServer.get(`/v2/buckets/${bucketId}/files/${file.id}/mirrors`)
+          .set('Authorization', getAuth(testUser))
+
+        // Assert
+        expect(response.status).toBe(200);
+        const body = response.body;
+
+        expect(body.bucket).toBe(bucketId)
+        expect(body.created).toBeDefined()
+        expect(body.index).toBe(index)
+        expect(body.shards).toHaveLength(2)
+
+        const [firstShard, secondShard] = body.shards
+
+        expect(firstShard.hash).toBeDefined()
```

Comment:
It could be defined, but null. We should check the value and the format of these data attributes